contracts/proof_market/mina_state_verifier.sol

```diff
@@ -0,0 +1,61 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts/access/Ownable.sol";
+import '@nilfoundation/evm-placeholder-verification/contracts/interfaces/verifier.sol';
+
+
+contract MinaStateVerifier is Ownable {
+    address _verifier;
+
+    address _base_gates;
+    address _scalar_gates;
+    uint256[][] _init_params;
+    int256[][][] _columns_rotations;
+
+    IVerifier v;
+
+    constructor(
+        address verifier,
+        address base_gates,
+        address scalar_gates,
+        uint256[][] memory init_params,
```

Comment:
We should fold this constructor code to the current verifier: https://github.com/NilFoundation/mina-state-proof/blob/master/contracts/state_proof/mina_state_proof.sol and update the interface.